### PR TITLE
[Merged by Bors] - chore(RingTheory/TensorProduct/Basic): Remove deprecated lemma

### PR DESCRIPTION
**Pull request by YaelDillies**
This was the only lemma needing the `LinearAlgebra.FiniteDimensional.Defs` import. Came up on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/lake.20exe.20graph.20--from.20Mathlib.2EGroupTheory.2EPerm.2ECycle.2EFactors)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
bors merge

**Comment by semorrison** [comment]
Oops, looks like it is failing further downstream

bors r-

File: import_graph.dot [hunk 1]
```diff
@@ -0,0 +1,2886 @@
+digraph "import_graph" {
```
Comment (semorrison) [inline import_graph.dot:1]: Please delete!

**Comment by semorrison** [comment]
bors d+

File: import_graph.dot (hunk 1 above)
Comment (YaelDillies) [reply import_graph.dot:1]: Raah, whoops

**Comment by YaelDillies** [comment]
bors merge

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**